mavisp/modules.py

```diff
@@ -680,6 +680,62 @@ class EnsembleSAS(MavispMultiEnsembleModule, module_class=TaccSAS):
     module_dir = "sas"
     name = "sas"
 
+
+
+class EFoldMine(MavispModule):
+
+    module_dir = "efoldmine"
+    name = "efoldmine"
+
+    def ingest(self, mutations):
+        warnings = []
+        efoldmine_file = os.listdir(os.path.join(self.data_dir, self.module_dir))
+        if len(efoldmine_file) != 1:
+            this_error = f"multiple or no files found in {efoldmine_file}; only one expected"
+            raise MAVISpMultipleError(warning=warnings,
+                                      critical=[MAVISpCriticalError(this_error)])
+
+        efoldmine_file = efoldmine_file[0]
+
+        log.info(f"parsing efoldmine file {efoldmine_file}")
+
+        try:
+            ef_res = pd.read_csv(os.path.join(self.data_dir, self.module_dir, efoldmine_file))
+        except Exception as e:
+            this_error = f"Exception {type(e).__name__} occurred when parsing the efoldmine file. Arguments:{e.args}"
+            raise MAVISpMultipleError(warning=warnings,
+                                        critical=[MAVISpCriticalError(this_error)])
+
+        # Filter only the required columns:
+        efoldmine_parsed = ef_res[['residue_index', 'earlyFolding']]
+
+
+
+        # Create an empty list to append the results:
+        result = []
+        
+        # Check if mutations correspond to early folding residues:
+        for mut in mutations:
+            mut_resn = int(mut[1:-1])
+            row = efoldmine_parsed[efoldmine_parsed['residue_index'] == mut_resn] 
+            if not row.empty and row['earlyFolding'].iloc[0] > 0.169:
+                prev_row = efoldmine_parsed[efoldmine_parsed['residue_index'] == mut_resn - 1]
+                next_row = efoldmine_parsed[efoldmine_parsed['residue_index'] == mut_resn + 1]
+                if not prev_row.empty and not next_row.empty and prev_row['earlyFolding'].iloc[0] > 0.169 and next_row['earlyFolding'].iloc[0] > 0.169:
+                    result.append((True, row['earlyFolding'].iloc[0]))
+                else:
+                    result.append((False, row['earlyFolding'].iloc[0])) 
+            else:
+                result.append((False, row['earlyFolding'].iloc[0]))  
```

Comment:
we should change the approach to:
  - we precompute on the sequence which residues are part of early folding regions and which aren't by using a sliding window approach
  - we then compare it with our mutations to determine which ones are in early folding regions